### PR TITLE
Added new test for decoupling logstash config from Operator

### PR DESCRIPTION
**Pull request by hzhao-github**
use cases:

1. Test when new variable createLogStashConfigMap sets to true, a configMap named weblogic-operator-logstash-cm
   is created and users can update logstash configuration by updating the configmap instead of rebuilding operator image.
2. Test that users can config and create their own configMap weblogic-operator-logstash-cm when new variable createLogStashConfigMap sets to false.

Jenkin Run passed

https://build.weblogick8s.org:8443/job/weblogic-kubernetes-operator-kind-new/9517/

### CODE REVIEW

File: integration-tests/src/test/java/oracle/weblogic/kubernetes/ItElasticLogging.java [hunk 1]
```diff
@@ -77,7 +90,7 @@
 /**
  * To test ELK Stack used in Operator env, this Elasticsearch test does
  * 1. Install Kibana/Elasticsearch.
- * 2. Install and start Operator with ELK Stack enabled.
+ * 2. Install and start Operators with ELK Stack enabled.
```
Comment (anpanigr) [inline integration-tests/src/test/java/oracle/weblogic/kubernetes/ItElasticLogging.java:93]: Describe the usage of  two Operator(s)

**Review by alai8 — APPROVED** [review]

File: integration-tests/src/test/java/oracle/weblogic/kubernetes/ItElasticLogging.java [hunk 2]
```diff
@@ -343,6 +366,69 @@ void testWlsLoggingExporter() throws Exception {
     logger.info("Query WebLogic log info succeeded");
   }
 
+  /**
+   * Test when new variable createLogStashConfigMap sets to true, a configMap named weblogic-operator-logstash-cm
```
Comment (anpanigr) [inline integration-tests/src/test/java/oracle/weblogic/kubernetes/ItElasticLogging.java:370]: Remove the word new

File: integration-tests/src/test/java/oracle/weblogic/kubernetes/ItElasticLogging.java [hunk 3]
```diff
@@ -343,6 +366,69 @@ void testWlsLoggingExporter() throws Exception {
...
+        "replaceStringInFile failed");
+    assertTrue(replaceConfigMap(opNamespace, configMapName, destConfigFile),
+        String.format("Failed to replace configMap %s in namespace %s", configMapName, opNamespace));
+
+    verifyLogstashConfigMapModifyResult(replaceStr);
+  }
+
+  /**
+   * Test that users can config and create their own configMap weblogic-operator-logstash-cm
+   * when new variable createLogStashConfigMap sets to false.
```
Comment (anpanigr) [inline integration-tests/src/test/java/oracle/weblogic/kubernetes/ItElasticLogging.java:404]: Remove the word new.
The description does not match the usecase implementation. Th test ends with a starting a second operator.  Does the test verify the custom configmap in the Operator namespace.

File: integration-tests/src/test/java/oracle/weblogic/kubernetes/ItElasticLogging.java [hunk 4]
```diff
@@ -473,4 +559,64 @@ private String execSearchQuery(String queryCriteria, String index) {
...
+      logger.info("Failed to exec the command command {0}. Error is {1} ", restartLogstashCmd, execResult.stderr());
+    }
+
+    // wait for logstash config modified and verify
+    withStandardRetryPolicy.untilAsserted(
+        () -> assertTrue(copyConfigFrPodAndSearchForString(containerName, replaceStr),
+            String.format("Failed to find search string %s", replaceStr)));
+  }
+
+  private boolean copyConfigFrPodAndSearchForString(String containerName, String replaceStr) {
```
Comment (anpanigr) [inline integration-tests/src/test/java/oracle/weblogic/kubernetes/ItElasticLogging.java:599]: copyConfigFrPodAndSearchForString --> copyConfigFromPodAndSearchForString

**Review by anpanigr — COMMENTED** [review]
The Jenkin run is not clean. 
Remove the @Disabled test in the class.

File: integration-tests/src/test/java/oracle/weblogic/kubernetes/ItElasticLogging.java (hunk 3 above)
Comment (hzhao-github) [reply integration-tests/src/test/java/oracle/weblogic/kubernetes/ItElasticLogging.java:404]: means the users are able to helm install the Operator with their own configMap weblogic-operator-logstash-cm

Comment (hzhao-github) [reply integration-tests/src/test/java/oracle/weblogic/kubernetes/ItElasticLogging.java:404]: I am going to change the description to

   * Test that users are able to use their own configMap weblogic-operator-logstash-cm
   * to create the Operator when variable createLogStashConfigMap sets to false.

File: integration-tests/src/test/java/oracle/weblogic/kubernetes/utils/ConfigMapUtils.java [hunk 5]
```diff
@@ -139,6 +142,116 @@ public static void createConfigMapForDomainCreation(String configMapName, List<P
...
+    List<V1ConfigMap> cmList = Kubernetes.listConfigMaps(nameSpace).getItems();
+    V1ConfigMap configMapToModify = cmList.stream()
+        .filter(cm -> cmName.equals(cm.getMetadata().getName()))
+        .findAny()
+        .orElse(null);
+
+    assertNotNull(configMapToModify,"Can't find cm for " + cmName);
+    Map<String, String> cmData = configMapToModify.getData();
+
+    String values = cmData.get("logstash.conf").replace(oldRegex,newRegex);
```
[the remark this answers is not in the document]
Comment (hzhao-github) [reply integration-tests/src/test/java/oracle/weblogic/kubernetes/utils/ConfigMapUtils.java:239]: changed

File: integration-tests/src/test/java/oracle/weblogic/kubernetes/ItElasticLogging.java (hunk 1 above)
Comment (hzhao-github) [reply integration-tests/src/test/java/oracle/weblogic/kubernetes/ItElasticLogging.java]: changed

File: integration-tests/src/test/java/oracle/weblogic/kubernetes/ItElasticLogging.java (hunk 2 above)
Comment (hzhao-github) [reply integration-tests/src/test/java/oracle/weblogic/kubernetes/ItElasticLogging.java]: removed

File: integration-tests/src/test/java/oracle/weblogic/kubernetes/ItElasticLogging.java (hunk 3 above)
Comment (hzhao-github) [reply integration-tests/src/test/java/oracle/weblogic/kubernetes/ItElasticLogging.java]: removed and changed

File: integration-tests/src/test/java/oracle/weblogic/kubernetes/ItElasticLogging.java (hunk 4 above)
Comment (hzhao-github) [reply integration-tests/src/test/java/oracle/weblogic/kubernetes/ItElasticLogging.java]: changed

**Review by anpanigr — APPROVED** [review]
lgtm

**Pull request merged**